### PR TITLE
Monito el

### PR DESCRIPTION
**Pull request by Elvisnando**
ecco

### CODE REVIEW

File: example/core.cpp [hunk 1]
```diff
@@ -23,4 +23,6 @@ void Core::onUdevEvent(UdevNotifier::UdevNotifier::Action action, UdevNotifier::
 {
     qDebug("[Core::onUdevEvent]");
     Q_UNUSED(device);
+
```
Comment (nwoki) [inline example/core.cpp]: remove unused whitespaeced

File: lib/device.h [hunk 2]
```diff
@@ -30,9 +30,10 @@ class UDEVNOTIFIERSHARED_EXPORT Device
     QString node() const;
     QString subsystem() const;
     QString type() const;
+    DevicePrivate * const d;
```
Comment (nwoki) [inline lib/device.h]: do **NOT** move the private pointer into the `public` area. This is **NOT** how you use d-pointers. If you want to access its members, do so via functions. If any needed functions are missing, add them to the base class

File: lib/lib.pro [hunk 3]
```diff
@@ -18,14 +18,16 @@ DEFINES += QT_DEPRECATED_WARNINGS
 
 SOURCES += \
     device.cpp \
-    udevnotifier.cpp
+    udevnotifier.cpp \
```
Comment (nwoki) [inline lib/lib.pro]: please keep alphabetical order

File: lib/lib.pro [hunk 4]
```diff
@@ -18,14 +18,16 @@ DEFINES += QT_DEPRECATED_WARNINGS
...
+    monitor.cpp
 
 HEADERS += \
     device.h \
     device_p.h \
     udevnotifier.h \
     udevnotifier_global.h \
-    udevnotifier_p.h
+    udevnotifier_p.h \
```
Comment (nwoki) [inline lib/lib.pro]: here too

File: lib/monitor.cpp [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    w->device = device;
+
+     if(w->type=="drm_minor"){
+
+          qDebug() << "Questò è un monitor" ;
+     }else
+ qDebug() << "Questò NON é monitor" ;
+
+}
+*/
```
Comment (nwoki) [inline lib/monitor.cpp]: remove commented code you're not using

File: lib/monitor.cpp [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+Monitor::Monitor(udev_device* device):Device(device)
+
+
+{
+
+       if(d->type=="drm_minor"){
+
+          qDebug() << "Questò è un monitor" ;
+     }else
+ qDebug() << "Questò NON é monitor" ;
```
Comment (nwoki) [inline lib/monitor.cpp]: Ugly

https://community.kde.org/Policies/Kdelibs_Coding_Style

File: lib/monitor.h [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+
+class udev_device;
+
+namespace UdevNotifier {
+
+class Monitor: public Device
+{
+
+public:
+
```
Comment (nwoki) [inline lib/monitor.h]: coding style here as well. Remove uneeded whitespaces

File: lib/udevnotifier.cpp [hunk 8]
```diff
@@ -5,7 +5,7 @@
 
 #include <libudev.h>
 #include <poll.h>
-
+#include<monitor.h>
```
Comment (nwoki) [inline lib/udevnotifier.cpp]: alphabetical order here too please

File: lib/udevnotifier.cpp [hunk 9]
```diff
@@ -88,20 +88,33 @@ void UdevNotifier::run()
             // XXX
 //             qDebug() << "hotplug[" << udev_device_get_action(dev) << "] " << udev_device_get_devnode(dev) << "," << udev_device_get_subsystem(dev) << "," << udev_device_get_devtype(dev);
             // emit the found device
-            Q_EMIT udevEvent(actionFromString(udev_device_get_action(dev)), new Device(dev));
+            QString det= udev_device_get_devtype(dev);
+         //   qDebug()<<det;
+            QString det1= "usb_device";
```
Comment (nwoki) [inline lib/udevnotifier.cpp]: no need to create seperate QString here. You can do the check in the if condition directly

File: lib/udevnotifier.cpp [hunk 10]
```diff
@@ -88,20 +88,33 @@ void UdevNotifier::run()
...
+            QString det2= "drm_minor";
+            QString det3= "usb_device";
+           // qDebug()<<det1;
+
+           if(udev_device_get_devtype(dev)==det2){
+           Q_EMIT udevEvent(actionFromString(udev_device_get_action(dev)), new Monitor(dev));}
```
Comment (nwoki) [inline lib/udevnotifier.cpp]: coding style

**Review by nwoki — COMMENTED** [review]
Please review my comments and apply modifications where requested

File: lib/udevnotifier.cpp [hunk 11]
```diff
@@ -1,18 +1,12 @@
...
 #include <libudev.h>
+#include<monitor.h>
 #include <poll.h>
-
-
 #include <QtCore/QDebug>
-
-
+#include "udevnotifier.h"
+#include "udevnotifier_p.h"
```
Comment (nwoki) [inline lib/udevnotifier.cpp:7]: non includerlo

File: lib/udevnotifier.cpp [hunk 12]
```diff
@@ -88,20 +82,23 @@ void UdevNotifier::run()
             // XXX
 //             qDebug() << "hotplug[" << udev_device_get_action(dev) << "] " << udev_device_get_devnode(dev) << "," << udev_device_get_subsystem(dev) << "," << udev_device_get_devtype(dev);
             // emit the found device
-            Q_EMIT udevEvent(actionFromString(udev_device_get_action(dev)), new Device(dev));
+           if(udev_device_get_devtype(dev)==QStringLiteral("drm_minor")){
```
Comment (nwoki) [inline lib/udevnotifier.cpp]: sistema indentazione e spaziature (see coding conventions)

File: lib/udevnotifier.cpp (hunk 11 above)
Comment (Elvisnando) [reply lib/udevnotifier.cpp:7]: se tolgo non compila

Comment (nwoki) [reply lib/udevnotifier.cpp:7]: My bad. Non avevo letto il file name

**Pull request merged**